docs/aws/audit/elbmonitoring/rules/min_ec2_instances_available_remediation.mdx

```diff
@@ -1,6 +1,259 @@
+5. If the number of instances is less than the minimum required (usually 2 for high availability), then it is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine and configure it with your AWS account credentials. You can do this by running the following commands:
+
+   Installation:
+   ```
+   pip install awscli
+   ```
+   Configuration:
+   ```
+   aws configure
+   ```
+   You will be prompted to enter your AWS Access Key ID, Secret Access Key, default region name, and default output format.
+
+2. List all the Load Balancers: Use the following command to list all the Elastic Load Balancers (ELBs) in your AWS account:
+
+   ```
+   aws elbv2 describe-load-balancers
+   ```
+   This command will return a JSON output with details of all the ELBs.
+
+3. Get the details of the instances attached to each ELB: For each ELB, use the following command to get the details of the instances attached to it:
+
+   ```
+   aws elbv2 describe-target-health --target-group-arn <TargetGroupArn>
+   ```
+   Replace `<TargetGroupArn>` with the ARN of the target group associated with the ELB. This command will return a JSON output with details of all the instances attached to the ELB.
+
+4. Count the number of instances: You can then parse the JSON output to count the number of instances attached to each ELB. If the number of instances is less than the minimum required, then it is a misconfiguration. You can use a Python script to automate this process. Here is a simple example:
+
+   ```python
+   import json
+   import subprocess
+
+   # Run the AWS CLI command to get the details of the instances attached to the ELB
+   result = subprocess.run(['aws', 'elbv2', 'describe-target-health', '--target-group-arn', '<TargetGroupArn>'], stdout=subprocess.PIPE)
+
+   # Parse the JSON output
+   data = json.loads(result.stdout)
+
+   # Count the number of instances
+   num_instances = len(data['TargetHealthDescriptions'])
+
+   # Check if the number of instances is less than the minimum required
+   if num_instances < MIN_NUM_INSTANCES:
+       print('Misconfiguration detected: Number of instances is less than the minimum required')
+   ```
+   Replace `<TargetGroupArn>` with the ARN of the target group associated with the ELB, and `MIN_NUM_INSTANCES` with the minimum number of instances required.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can install it using pip:
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials. Replace 'your_access_key', 'your_secret_key', and 'your_region' with your actual AWS credentials and the region you want to check.
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='your_access_key',
+    aws_secret_access_key='your_secret_key',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>